libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CoreBotAdapter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// CoreBotAdapter is a base BotAdapter that derives from CloudAdapter and implements OnTurnError handling. This class
+    /// also registers any middleware services that are contained in the middlewares collection.
+    /// </summary>
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes. This class is only instantiated via dependency injection so no direct reference is made.
+    internal class CoreBotAdapter : CloudAdapter
+#pragma warning restore CA1812 // Avoid uninstantiated internal classes
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoreBotAdapter"/> class.
+        /// </summary>
+        /// <param name="botFrameworkAuthentication">An instance of a BotFrameworkAuthentication.</param>
+        /// <param name="middlewares">The collection of middleware instances.</param>
+        /// <param name="logger">The logger instance.</param>
+        public CoreBotAdapter(
+            BotFrameworkAuthentication botFrameworkAuthentication,
+            IEnumerable<IMiddleware> middlewares,
```

Comment:
As it is, the only way a dev could register middlewares for the bot would be to register a DI service with the if type `IEnumerable<IMiddleware>`